src/test-apps/Makefile.am

```diff
@@ -24,6 +24,8 @@
 
 include $(abs_top_nlbuild_autotools_dir)/automake/pre.am
 
+AM_LOG_DRIVER_FLAGS := --build-num $(shell echo $$PPID)
```

Comment:
I suspect this may blow up on shells in which SHELL is /bin/sh but in which /bin/sh is a symlink to /bin/dash or /bin/zsh rather than /bin/bash since I believe PPID is a bashism.

Specifying SHELL := `which bash` may work around this.

Reply:
tested with `echo $PPID` on dash, zsh, ash (which should be available on busybox systems), ksh, and multiple versions of bash. Worked in each case. Fails as might be expected in csh and tcsh, but on csh-based shells other things might fail as well (e.g. redirects that we use extensively).

Reply:
Changed the shell to `/bin/bash` -- there are a number of other components of the build system that depend on bash being installed in `/bin/bash` rather than in an arbitrary location.  I have also added a restriction on the length of the PID -- I now only take the last 5 characters plus a newline; this ensures that the overall network namespace ID continues to fit in 15 characters.